### ISSUE TITLE
E3002 false positives with intrinsics returning complex types when referenced via select

### ISSUE THREAD

**Issue by bgardner-noggin** [aws-cloudformation/cfn-lint#2707]
### CloudFormation Lint Version

0.76.2

### What operating system are you using?

Fedora

### Describe the bug

When using intrinsic functions and !Select to retrieve complex (ie structured) data from custom resources for other resources E3002 incorrectly marks the intrinsic as invalid properties of the resource.

For example, given a custom resource which returns an AWS::Lambda::Function Code object:

```
"Code": {
  [
   "S3Bucket": "<name>",
   "S3Key": "<object-key>",
   "S3ObjectVersion": "<object-version>"
 ],
 [
   "S3Bucket": "<name 2>",
   "S3Key": "<object-key 2>",
   "S3ObjectVersion": "<object-version 2>"
 ]
} 
```

Then referencing this as the Code property for the Function resource via select eg

`Code: !Select [ 0, !GetAtt CustomResource.Output.Code ]`

cfn-lint raises an E3002 of the form:
E3002 Invalid Property Resources/Function/Properties/Code/Fn::Select

This is similar to https://github.com/aws-cloudformation/cfn-lint/issues/2537 which resolved a similar case.

### Expected behavior

E3002 should not be raised when referencing via !Select

### Reproduction template

```yaml
Parameters:
    RoleArn:
        Type: String
    FunctionArn:
        Type: String

Resources:

    GetCodeForLatestS3Object:
        Type: Custom::LambdaSource
        Properties:
            ServiceToken: !Ref FunctionArn
            S3Url: "s3://some-bucket/lambdazips/helper.zip"

    Helper:
        Type: 'AWS::Lambda::Function'
        Properties:
            Handler: 'helper.lambda_handler'
            Role: !Ref RoleArn
            Code: !Select [ 0, !GetAtt GetCodeForLatestS3Object.Code ]
```
